ocaml/message-switch/core/make.ml

```diff
@@ -352,33 +352,43 @@ functor
 
     let listen ~process ~switch:port ~queue:name () =
       let token = Printf.sprintf "%d" (Unix.getpid ()) in
-      M.connect port >>= fun c ->
-      Connection.rpc c (In.Login token) >>|= fun (_ : string) ->
-      Connection.rpc c (In.CreatePersistent name) >>|= fun _ ->
+      let reconnect () =
+        M.connect port >>= fun request_conn ->
+        Connection.rpc request_conn (In.Login token) >>|= fun (_ : string) ->
+        Connection.rpc request_conn (In.CreatePersistent name) >>|= fun _ ->
+        M.connect port >>= fun reply_conn ->
+        Connection.rpc reply_conn (In.Login token) >>|= fun (_ : string) ->
+        Connection.rpc reply_conn (In.CreatePersistent name) >>|= fun _ ->
```

Comment:
The old code didn't seem to create a persistent one on reply conn, and the Protocol_unix doesn't either (only on reuqest, not on reply).

Reply:
makes sense